Sources/DarwinPlatform/Darwin.swift

```diff
@@ -0,0 +1,144 @@
+import Foundation
+import SwiftlyCore
+@_implementationOnly import Archive
+
+/// `Platform` implementation for Darwin systems.
+/// This implementation can be reused for any supported Darwin platform.
+/// TODO: replace dummy implementations
+public struct Darwin: Platform {
+    public init() {}
+
+    public var appDataDirectory: URL {
+        if let dir = ProcessInfo.processInfo.environment["XDG_DATA_HOME"] {
+            return URL(fileURLWithPath: dir)
+        } else {
+            return FileManager.default.homeDirectoryForCurrentUser
+                .appendingPathComponent(".local", isDirectory: true)
+                .appendingPathComponent("share", isDirectory: true)
+        }
+    }
+
+    public var toolchainFileExtension: String {
+        "pkg"
+    }
+
+    public func isSystemDependencyPresent(_: SystemDependency) -> Bool {
+        true
+    }
+
+    public func install(from tmpFile: URL, version: ToolchainVersion) throws {
+        guard tmpFile.fileExists() else {
+            throw Error(message: "\(tmpFile) doesn't exist")
+        }
+
+        if !self.swiftlyToolchainsDir.fileExists() {
+            try FileManager.default.createDirectory(at: self.swiftlyToolchainsDir, withIntermediateDirectories: false)
+        }
+
+        SwiftlyCore.print("Extracting toolchain...")
+        let toolchainDir = self.swiftlyToolchainsDir.appendingPathComponent(version.name)
+
+        if toolchainDir.fileExists() {
+            try FileManager.default.removeItem(at: toolchainDir)
+        }
+
+        try extractArchive(atPath: tmpFile) { name in
```

Comment:
Does this work on `pkg` files

Reply:
Not sure.
The same reason as last one. https://github.com/swift-server/swiftly/pull/62#discussion_r1331905196